src/main/java/de/zalando/aruha/nakadi/repository/db/EventTypeCachedRepository.java

```diff
@@ -0,0 +1,49 @@
+package de.zalando.aruha.nakadi.repository.db;
+
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.exceptions.NakadiException;
+import de.zalando.aruha.nakadi.exceptions.NoSuchEventTypeException;
+import de.zalando.aruha.nakadi.repository.EventTypeRepository;
+
+import java.util.List;
+
+public class EventTypeCachedRepository implements EventTypeRepository {
+
+    private final EventTypeRepository repository;
+
+    private final EventTypeCache cache;
+
+    public EventTypeCachedRepository(EventTypeRepository repository,
```

Comment:
This a "decorated" version of the repository, in the sense that it just adds some functionality on top of the DB repo without changing the interface.

I also isolated all caching behaviour in the cache class, not to mess up with things here.

I thought this approach would be a less intrusive way of adding the caching. But feel free to make drastic suggestions of design change.